components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1411,454 @@ private static Validator resourceToValidatorObject(
+                            (IdentityTenantUtil.getTenantId(tenantDomain), X509_CA_CERT_RESOURCE_TYPE, CERTS);
+            List<ResourceFile> resourceFiles = resource.getFiles();
+            if (resourceFiles == null || resourceFiles.isEmpty()) {
+                log.warn("Resource files are empty for certificates in tenant: " + resource.getTenantDomain());
+                return null;
+            }
+
+            ResourceFile resourceFile = resourceFiles.get(0);
+            if (resourceFile == null) {
+                log.warn("Resource file is null for certificates in tenant: " + resource.getTenantDomain());
+                return null;
+            }
+
+            InputStream inputStream = CertValidationDataHolder.getInstance()
+                    .getConfigurationManager()
+                    .getFileById(X509_CA_CERT_RESOURCE_TYPE, CERTS, resourceFile.getId());
+
+            if (inputStream == null) {
+                log.warn("InputStream is null for the file in resource.");
+                return null;
+            }
+
+            String fileContent = convertInputStreamToString(inputStream);
+            IssuerDNMap issuerDNMap = ModelSerializer.deserializeIssuerDNMap(fileContent);
+
+            for (Map.Entry<String, List<CertObject>> entry : issuerDNMap.getIssuerCertMap().entrySet()) {
+                List<CertObject> certList = entry.getValue();
+                Optional<CertObject> certObject = certList.stream()
+                        .filter(cert -> cert.getCertId().equals(certificateId))
+                        .findFirst();
+
+                if (certObject.isPresent()) {
+                    return new CACertificate(certObject.get().getCrlUrls(),
+                            certObject.get().getOcspUrls(),
+                            getCACertificateFromCertificateManager(certificateId, tenantDomain));
+                }
+            }
+            return null;
+        } catch (ConfigurationManagementException e) {
+            throw new CertificateValidationException("Error while processing the resource", e);
+        } catch (IOException e) {
+            throw new CertificateValidationException("Error while reading the file content", e);
+        }
+    }
+
+    private static X509Certificate getCACertificateFromCertificateManager(String certificateId,
+                                                                         String tenantDomain)
+            throws CertificateValidationException {
+
+        try {
+            Certificate certificate = CertValidationDataHolder.getInstance()
+                    .getCertificateManagementService()
+                    .getCertificate(certificateId, tenantDomain);
+
+            return decodeCertificate(certificate.getCertificateContent());
+        } catch (CertificateMgtException e) {
+            log.debug("Error when getting certificate from certificate manager.", e);
```

Comment:
no need to log the error here since the exception is thrown.

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59